src/CommunicationExecutor.c

```diff
@@ -353,7 +409,19 @@ void Ros_Communication_StartExecutors(SEM_ID semCommunicationExecutorStatus)
     //===========================================================
     //===========================================================
     //===========================================================
-    
+
+    // See if we should start task which monitors user lan port link state
+    // (This task deletes itself when the agent disconnects.)
+    if (g_nodeConfigSettings.userlan_monitor_enabled)
+    {
+        Ros_Debug_BroadcastMsg("Starting UserLan link state monitor task ..");
+        mpCreateTask(MP_PRI_TIME_NORMAL, MP_STACK_SIZE,
```

Comment:
I'm a little concerned about starting up another task. I'm worried that the all the task switching may be contributing to delays/latency.

Could we put this functionality in `main.c` inside the loop in `RosInitTask`?

Reply:
My concern is based on a little bit more than assumption. I have empirically measured delays with spooling up too many tasks for another project. (Granted, that situation had many more tasks than MotoROS2.)

While the motion task is indeed at a higher priority, a `normal` priority task is the lowest priority of anything else in the robot controller. 

You're _probably_ right that this one task won't cause noticeable delay. I'm just nervous because we frequently have users wanting "maximum" speed and zero latency.

Reply:
> My concern is based on a little bit more than assumption. I have empirically measured delays with spooling up too many tasks for another project. (Granted, that situation had many more tasks than MotoROS2.)

Would you happen to know what sort of task scheduling is configured on the controller?

I could see round-robin resulting in significantly increased delays with a high nr of tasks.

> While the motion task is indeed at a higher priority, a `normal` priority task is the lowest priority of anything else in the robot controller.

So are you worried the monitoring task might get starved? Or am I misunderstanding you?

> You're _probably_ right that this one task won't cause noticeable delay. I'm just nervous because we frequently have users wanting "maximum" speed and zero latency.

Sure.

I'll see if I can setup some tracing/measurement infrastructure and compare.

Reply:
> Would you happen to know what sort of task scheduling is configured on the controller?

From M+ doc:

> The task scheduling method is preemptive scheduling based on priority.
The tasks in the Ready state with the same priority are executed in a timeshared manner by round robin scheduling.

------------
> So are you worried the monitoring task might get starved? Or am I misunderstanding you?

I'm worried that all the tasks (publishers, executor, services, etc) would be starved and reduce the "responsiveness".

Reply:
> Are we sure a single additional task running at 2 Hz will cause problems?

No. But I'm very paranoid about such things. 

> https://github.com/Yaskawa-Global/motoros2/commit/c59e1b8e132bcdbfd7d9eb561d87031351340748 changes the task into a rcl_timer_t callback

I like this idea